### PR TITLE
LibWeb+LibJS: Clear exceptions in each Document::run_javascript() call / Assert when exception is not cleared before Interpreter::run()

### PR DESCRIPTION
**Pull request by linusg**
**LibWeb: Clear exceptions in each Document::run_javascript() call**

We don't want to carry over exceptions across multiple `Document::run_javascript()` calls as `Interpreter::run()` and every of its exception checks will get confused - in this case there would be an exception, but not because a certain action failed.

Real-life example:

```html
<script>var a = {}; a.test()</script>
<script>alert("It worked!")</script>
```

The above HTML will invoke `Document::run_javascript()` twice, the first call will result in a `TypeError`, which is still stored during the second call. The interpreter will eventually call the following functions (in order) for the `alert()` invocation:

- `Identifier::execute()`
- `Interpreter::get_variable()`
- `Object::get()` (on the global object)

That last `Object::get()` call has an exception check which is triggered as we still carry around the exception from earlier - and eventually returns an empty value.

Long story short, the second script will wrongly fail with "ReferenceError, 'alert' is not defined".

**LibJS: Assert when exception is not cleared before Interpreter::run()**

This is to prevent bugs like #3091 (fixed in 04d47ea) in the future; we generally
don't want `Interpreter::run()` to be called if the interpreter still has an exception stored. Sure, it could clear those itself but letting users of the interpreter do it explicitly seems sensible.

### CODE REVIEW

**Comment by Lubrsi** [comment]
Confirmed fixed 👍 . This also happened across external scripts so loading a site such as Twitter had a bunch of weird exceptions thrown such as "window" is not defined.

File: Libraries/LibWeb/DOM/Document.cpp [hunk 1]
```diff
@@ -436,6 +436,7 @@ JS::Value Document::run_javascript(const StringView& source)
         parser.print_errors();
         return JS::js_undefined();
     }
+    document().interpreter().clear_exception();
     return document().interpreter().run(document().interpreter().global_object(), *program);
```
Comment (awesomekling) [inline Libraries/LibWeb/DOM/Document.cpp:440]: Wouldn't it be nicer to do an exception check *after* running code, and then clearing it?

Comment (linusg) [reply Libraries/LibWeb/DOM/Document.cpp]: yeah that makes sense :)

**Comment by awesomekling** [comment]
Great fix. Good idea with the assertion too
